openlibrary/plugins/importapi/code.py

```diff
@@ -63,8 +63,10 @@ def parse_data(data):
     :rtype: (dict|None, str|None)
     :return: (Edition record, format (rdf|opds|marcxml|json|marc)) or (None, None)
     """
+    if six.py3 and isinstance(data, bytes):
+        data = data.decode('utf-8')
     data = data.strip()
```

Comment:
This `strip()` looks wrong/ redundant- it appears to be stripping POSTed data of currently undetermined type. Unsure what real value that provides even when it triggers on XML or JSON data. If this is the line that is erroring on type I think it can be safely removed.

Reply:
`str.strip()` and `bytes.strip()` both work as expected regardless of encoding so I am not inclined to remove this from a function that needs to accept both bytes and str but has no tests.

 % `python3`
```
>>> ten_spaces = " " * 10
>>> payload = ten_spaces + "Lüsai" + ten_spaces
>>> payload.strip()
'Lüsai'
>>> payload.encode("gb18030").strip()
b'L\xa8\xb9sai'
>>> payload.encode("latin-1").strip()
b'L\xfcsai'
>>> payload.encode("utf-8").strip()
b'L\xc3\xbcsai'
```